clients/java/src/main/java/io/zeebe/client/impl/worker/JobClientImpl.java

```diff
@@ -51,15 +52,33 @@ public CompleteJobCommandStep1 newCompleteCommand(final long jobKey) {
         asyncStub, objectMapper, jobKey, config.getDefaultRequestTimeout(), retryPredicate);
   }
 
+  @Override
+  public CompleteJobCommandStep1 newCompleteCommand(ActivatedJob job) {
+    return new CompleteJobCommandImpl(
```

Comment:
As I wrote earlier I think this is not necessary. Either we delegate in ZeebeClient directly to the method which accepts the jobkey or we do it here and just call newCompletecommand(jobKey)